banzai/qc/header_checker.py

```diff
@@ -170,17 +191,28 @@ def check_exptime_value(self, image):
 
         """
 
-        exptime_value = image.header['EXPTIME']
-
         logging_tags = logs.image_config_to_tags(image, self.group_by_keywords)
 
-        if exptime_value < 0.0:
-            sentence = 'The header EXPTIME key got the unexpected value : negative value'
-            self.logger.error(sentence, extra=logging_tags)
+        try:
+            exptime_value = image.header['EXPTIME']
+            self.save_qc_results({"header_exptime": exptime_value}, image)
+        except:
             return
 
-        obstype = image.header['OBSTYPE']
-        if (exptime_value == 0.0) & (obstype == 'EXPOSE'):
-            sentence = 'The header EXPTIME key got the unexpected value : 0.0'
-            self.logger.error(sentence, extra=logging_tags)
-            return
+        try:
+            exptime_negative = exptime_value < 0.0
+            if exptime_negative:
+                sentence = 'The header EXPTIME key got the unexpected value : negative value'
+                self.logger.error(sentence, extra=logging_tags)
+            self.save_qc_results({"HeaderExptimeNegative": exptime_negative}, image)
+        except:
+            pass
+
+        try:
+            exptime_zero = (exptime_value == 0.0) & (image.header['OBSTYPE'] == 'EXPOSE')
+            if exptime_zero:
+                sentence = 'The header EXPTIME key got the unexpected value : 0.0'
+                self.logger.error(sentence, extra=logging_tags)
+            self.save_qc_results({"HeaderExptimeZero": exptime_zero}, image)
+        except:
```

Comment:
More specific?

Reply:
The header_checker changes shouldn't have ended up in this PR. That class still needs work that @turnerm is going to do. As such, I'm pulling out of the main code path for the moment and will have her open a new PR with those changes. Will take these comments into account in the other PR.